src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/PersonNeedsTest.kt

```diff
@@ -0,0 +1,215 @@
+                        .withSection("STANDARD_HARM")
+                        .withName("STANDARD_HARM")
+                        .withOverThreshold(true)
+                        .withRiskOfHarm(true)
+                        .withRiskOfReoffending(false)
+                        .withFlaggedAsNeed(true)
+                        .withIdentifiedAsNeed(true)
+                        .withNeedScore(5)
+                        .produce(),
+                      NeedFactory()
+                        .withSeverity(NeedSeverity.STANDARD)
+                        .withSection("RISK_OF_REOFFEND")
+                        .withName("RISK_OF_REOFFEND")
+                        .withOverThreshold(true)
+                        .withRiskOfHarm(false)
+                        .withRiskOfReoffending(true)
+                        .withFlaggedAsNeed(true)
+                        .withIdentifiedAsNeed(true)
+                        .withNeedScore(5)
+                        .produce()
+                    )
+                  ).produce()
+              )
+            )
+        )
+    )
+
+    val jwt = jwtAuthHelper.createAuthorizationCodeJwt(
+      subject = "username",
+      authSource = "delius",
+      roles = listOf("ROLE_PROBATION")
+    )
+
+    webTestClient.get()
+      .uri("/people/CRN/needs")
+      .header("Authorization", "Bearer $jwt")
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBody()
+      .json(
+        objectMapper.writeValueAsString(
+          PersonNeeds(
+            linkedToRiskOfSeriousHarm = listOf(
+              PersonNeed(
+                section = "SEVERE_HARM",
+                name = "SEVERE_HARM",
+                overThreshold = true,
+                riskOfHarm = true,
+                flaggedAsNeed = true,
+                severity = "SEVERE",
+                identifiedAsNeed = true,
+                needScore = 5,
+                riskOfReoffending = true
+              )
+            ),
+            linkedToReoffending = listOf(
+              PersonNeed(
+                section = "RISK_OF_REOFFEND",
+                name = "RISK_OF_REOFFEND",
+                overThreshold = true,
+                riskOfHarm = false,
+                flaggedAsNeed = true,
+                severity = "STANDARD",
+                identifiedAsNeed = true,
+                needScore = 5,
+                riskOfReoffending = true
+              )
+            ),
+            notLinkedToSeriousHarmOrReoffending = listOf(
+              PersonNeed(
+                section = "STANDARD_HARM",
+                name = "STANDARD_HARM",
+                overThreshold = true,
+                riskOfHarm = true,
+                flaggedAsNeed = true,
+                severity = "STANDARD",
+                identifiedAsNeed = true,
+                needScore = 5,
+                riskOfReoffending = false
+              )
+            )
+          )
+        )
+      )
+  }
+}
```

Comment:
this is really easy to follow. I wonder if this last test could be a little easier to parse if you used the `needScore` as hook to link the fixtures with the assertions? e.g.

```
 wiremockServer.stubFor(
      get(WireMock.urlEqualTo("/needs/crn/CRN"))
        .willReturn(
          aResponse()
            .withHeader("Content-Type", "application/json")
            .withStatus(200)
            .withBody(
              objectMapper.writeValueAsString(
                NeedsFactory()
                  .withAssessedOn(LocalDateTime.now())
                  .withIdentifiedNeeds(
                    listOf(
                      NeedFactory()
                        .withSeverity(NeedSeverity.SEVERE)
                        .withSection("SEVERE_HARM")
                        .withName("SEVERE_HARM")
                        .withOverThreshold(true)
                        .withRiskOfHarm(true)
                        .withRiskOfReoffending(true)
                        .withFlaggedAsNeed(true)
                        .withIdentifiedAsNeed(true)
                        .withNeedScore(10)
                        .produce(),
                      NeedFactory()
                        .withSeverity(NeedSeverity.STANDARD)
                        .withSection("STANDARD_HARM")
                        .withName("STANDARD_HARM")
                        .withOverThreshold(true)
                        .withRiskOfHarm(true)
                        .withRiskOfReoffending(false)
                        .withFlaggedAsNeed(true)
                        .withIdentifiedAsNeed(true)
                        .withNeedScore(2)
                        .produce(),
                      NeedFactory()
                        .withSeverity(NeedSeverity.STANDARD)
                        .withSection("RISK_OF_REOFFEND")
                        .withName("RISK_OF_REOFFEND")
                        .withOverThreshold(true)
                        .withRiskOfHarm(false)
                        .withRiskOfReoffending(true)
                        .withFlaggedAsNeed(true)
                        .withIdentifiedAsNeed(true)
                        .withNeedScore(5)
                        .produce()
                    )
                  ).produce()
              )
            )
        )
    )

    val jwt = jwtAuthHelper.createAuthorizationCodeJwt(
      subject = "username",
      authSource = "delius",
      roles = listOf("ROLE_PROBATION")
    )

    webTestClient.get()
      .uri("/people/CRN/needs")
      .header("Authorization", "Bearer $jwt")
      .exchange()
      .expectStatus()
      .isOk
      .expectBody()
      .json(
        objectMapper.writeValueAsString(
          PersonNeeds(
            linkedToRiskOfSeriousHarm = listOf(
              PersonNeed(
                section = "SEVERE_HARM",
                name = "SEVERE_HARM",
                overThreshold = true,
                riskOfHarm = true,
                flaggedAsNeed = true,
                severity = "SEVERE",
                identifiedAsNeed = true,
                needScore = 10,
                riskOfReoffending = true
              )
            ),
            linkedToReoffending = listOf(
              PersonNeed(
                section = "RISK_OF_REOFFEND",
                name = "RISK_OF_REOFFEND",
                overThreshold = true,
                riskOfHarm = false,
                flaggedAsNeed = true,
                severity = "STANDARD",
                identifiedAsNeed = true,
                needScore = 5,
                riskOfReoffending = true
              )
            ),
            notLinkedToSeriousHarmOrReoffending = listOf(
              PersonNeed(
                section = "STANDARD_HARM",
                name = "STANDARD_HARM",
                overThreshold = true,
                riskOfHarm = true,
                flaggedAsNeed = true,
                severity = "STANDARD",
                identifiedAsNeed = true,
                needScore = 2,
                riskOfReoffending = false
              )
            )
          )
        )
```

Reply:
👍